src/hubverse_transform/model_output.py

```diff
@@ -70,23 +69,26 @@ def from_s3(cls, bucket_name: str, s3_key: str, origin_prefix: str = "raw") -> "
     def parse_file(cls, file_name: str) -> dict:
         """Parse model-output file name into individual parts."""
 
-        # TODO: verify assumptions about format of model-output filenames
-        # Code below assumes [round_id likely yyyy-mm-dd]-[team]-[model] AND
-        # that there are no hyphens in team or model names
-        # https://github.com/orgs/Infectious-Disease-Modeling-Hubs/discussions/10
-        file_name_split = file_name.rsplit("-", 2)
+        # In practice, Hubverse hubs are formatting round_id as dates in YYYY-MM-DD format.
+        # There's an open discussion about whether or not we want to support round_ids in other
+        # formats, but until there's a final decision, this code will assume that each model-output
+        # file begins with a YYYY-MM-DD round_id.
+        # https://github.com/Infectious-Disease-Modeling-Hubs/hubValidations/discussions/13
 
-        if (
-            file_name.count("-") > 4
-            or len(file_name_split) != 3
-            or not re.match(r"^\d{4}-\d{2}-\d{2}$", file_name_split[0])
-        ):
-            raise ValueError(f"File name {file_name} not in expected model-output format: yyyy-mm-dd-team-model.")
+        round_id_match = re.match(r"^\d{4}-\d{2}-\d{2}", file_name)
+        if not round_id_match:
+            raise ValueError(f"Unable to get YYYY-MM-DD round_id from file name {file_name}.")
+        round_id = round_id_match.group(0)
+
+        # model_id is anything that comes after round_id in the filename
+        model_id_split = re.split(rf"{round_id}[-_]*", file_name)
+        if not model_id_split or len(model_id_split) <= 1 or not model_id_split[-1]:
+            raise ValueError(f"Unable to get model_id from file name {file_name}.")
+        model_id = "".join(model_id_split[-1].split())
```

Comment:
this removes any spaces in whatever is remaining. Is that your intent? So `2022-12-31-my team-my model.parquet` would return `"myteam-mymodel"` for `model_id`

Reply:
Good call-out. It was my intent at the time I made the pull request, but I have another in the queue (started after Wednesday's dev meeting) that retains the spaces and adds some corresponding test cases.